File: sorts/insertion_sort.py
def insertionSort(arr):
	n = len(arr)
	
	if n <= 1:
		return 

	for i in range(1, n): 
		current = arr[i] 
		j = i-1
		while j >= 0 and current < arr[j]: 
			arr[j+1] = arr[j] 
			j -= 1
		arr[j+1] = current 


arr = [12, 11, 13, 5, 6]
insertionSort(arr)
print(arr)

File: sorts/insertion_sort.cpp
#include <bits/stdc++.h>
using namespace std;

void insert_sort(int arr[], int n)
{
    for (int j = 1; j < n; j++)
    {
        int current = arr[j];
        int i = j - 1;

        while (i >= 0 && arr[i] > current)
        {
            arr[i + 1] = arr[i];
            i--;
        }
        arr[i + 1] = current;
    }

    for (int i = 0; i < n; i++)
    {
        cout << arr[i] << " ";
    }
    cout << endl;
}

int main()
{
    int arr[5] = {12, 3, 8, 9, 20};
    insert_sort(arr, 5);
    return 0;
}